File: exer_cpp/Cursos_CodCad/estruturas_neps/basicas/chaves.cpp
#include <bits/stdc++.h>

using namespace std;

stack<char> p;
string linha;
int n;
bool ok=true;
int main(){
    cin>> n;
    cin.ignore();
    while(n--){
        getline(cin,linha);
        for(char c: linha){
            if(c=='{'){
                p.push(c);
            }
            else if(c=='}'){
                if(!p.empty()){
                    p.pop();
                }
                else{
                    ok=false;
                }
            }
        }
    }
    if(!p.empty()){
        ok=false;
    }
    if(ok){
        cout<< 'S';
    }
    else{
        cout<<'N';
    }
    return 0;
}

File: exer_python/OBI/chaves.py
n = int(input())
P=[]
teste= True
for i in range(n):
    linha= input()
    #print("--> ",linha)
    if '{' in linha or '}' in linha:
        for i in range(len(linha)):
            if linha[i]=='{':
                P.append('{')
            elif linha[i]=='}':
                if P==[]:
                    teste= False                 
                    break   
                else:
                    P.pop()               
            
if (teste) and P==[]:
    print('S')
else:
    print('N')
